src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/Tranche.java

```diff
@@ -0,0 +1,260 @@
+    public final double targetTruthSensitivity;
+    public final double minVQSLod;  //minimum value of VQSLOD in this tranche
+    public final double knownTiTv;  //titv value of known sites in this tranche
+    public final double novelTiTv;  //titv value of novel sites in this tranche
+    public final int numKnown;      //number of known sites in this tranche
+    public final int numNovel;      //number of novel sites in this tranche
+    public final String name;       //Name of the tranche
+    public final VariantRecalibratorArgumentCollection.Mode model;    //this is a SNP VQSR tranche or Indel tranche?
+
+    private final int accessibleTruthSites;
+    private final int callsAtTruthSites;
+
+    public Tranche(double targetTruthSensitivity, double minVQSLod, int numKnown, double knownTiTv, int numNovel, double novelTiTv, int accessibleTruthSites, int callsAtTruthSites, VariantRecalibratorArgumentCollection.Mode model) {
+        this(targetTruthSensitivity, minVQSLod, numKnown, knownTiTv, numNovel, novelTiTv, accessibleTruthSites, callsAtTruthSites, model, "anonymous");
+    }
+
+    public Tranche(double targetTruthSensitivity, double minVQSLod, int numKnown, double knownTiTv, int numNovel, double novelTiTv, int accessibleTruthSites, int callsAtTruthSites, VariantRecalibratorArgumentCollection.Mode model, String name) {
+        if ( targetTruthSensitivity < 0.0 || targetTruthSensitivity > 100.0) {
+            throw new UserException("Target FDR is unreasonable " + targetTruthSensitivity);
+        }
+
+        if ( numKnown < 0 || numNovel < 0) {
+            throw new GATKException("Invalid tranche - no. variants is < 0 : known " + numKnown + " novel " + numNovel);
+        }
+
+        if ( name == null ) {
+            throw new GATKException("BUG -- name cannot be null");
+        }
+
+        this.targetTruthSensitivity = targetTruthSensitivity;
+        this.minVQSLod = minVQSLod;
+        this.novelTiTv = novelTiTv;
+        this.numNovel = numNovel;
+        this.knownTiTv = knownTiTv;
+        this.numKnown = numKnown;
+        this.model = model;
+        this.name = name;
+
+        this.accessibleTruthSites = accessibleTruthSites;
+        this.callsAtTruthSites = callsAtTruthSites;
+    }
+
+    public static final Comparator<Tranche> TRUTH_SENSITIVITY_ORDER = (tranche1, tranche2) -> Double.compare(tranche1.targetTruthSensitivity, tranche2.targetTruthSensitivity);
+
+    @Override
+    public String toString() {
+        return String.format("Tranche targetTruthSensitivity=%.2f minVQSLod=%.4f known=(%d @ %.4f) novel=(%d @ %.4f) truthSites(%d accessible, %d called), name=%s]",
+                targetTruthSensitivity, minVQSLod, numKnown, knownTiTv, numNovel, novelTiTv, accessibleTruthSites, callsAtTruthSites, name);
+    }
+
+    private static double getRequiredDouble(Map<String,String> bindings, String key ) {
+        if ( bindings.containsKey(key) ) {
+            return Double.valueOf(bindings.get(key));
```

Comment:
Wrap the potential `NumberFormatException` within either a `UserException` or `GATKException`, depending on whether or not the data came from user input.